rts/motoko-rts/src/gc/mark_compact/bitmap.rs

```diff
@@ -90,36 +151,37 @@ pub const BITMAP_ITER_END: u32 = 1024 * 1024 * 1024;
 impl BitmapIter {
     /// Returns the next bit, or `BITMAP_ITER_END` if there are no more bits set.
     pub fn next(&mut self) -> u32 {
-        debug_assert!(self.current_word_idx <= self.size);
+        debug_assert!(self.current_bit_idx <= self.size);
 
         // Outer loop iterates 64-bit words
         loop {
-            if self.current_word == 0 && self.current_word_idx == self.size {
+            if self.current_word == 0 && self.current_bit_idx == self.size {
                 return BITMAP_ITER_END;
             }
 
             // Inner loop iterates bits in the current word
```

Comment:
<img width="534" alt="BitmapIterator-next" src="https://user-images.githubusercontent.com/1312006/143659714-04f8c0e0-d395-4cfa-9450-acaf5d25654b.png">
The inner loop is a few instructions shorter, and has only `store` traffic.